fbpic/lpa_utils/laser/transverse_laser_profiles.py

```diff
@@ -0,0 +1,517 @@
+            GPU. This is usually the case if it only uses standard arithmetic
+            and numpy operations. Default: False.
+        """
+        assert propagation_direction in [-1, 1]
+        self.propag_direction = float(propagation_direction)
+        self.gpu_capable = gpu_capable
+
+    def evaluate(self, x, y, z):
+        """
+        Return the complex longitudinal laser profile.
+
+        Parameters
+        -----------
+        x, y, z: ndarrays (meters)
+            The position at which to calculate the profile (in the lab frame)
+
+        Returns:
+        --------
+        profile: ndarray
+            Arrays of the same shape as x, containing the complex
+            transverse profile
+        """
+        # The base class only defines dummy fields
+        # (This should be replaced by any class that inherits from this one.)
+        return np.zeros_like(x, dtype='complex')
+
+
+# Particular classes for each transverse laser profile
+# ------------------------------------------------------
+
+class GaussianTransverseProfile(LaserTransverseProfile):
+    """Class that calculates a Gaussian transverse laser profile."""
+
+    def __init__(self, waist, zf=0., lambda0=0.8e-6, propagation_direction=1):
+        """
+        Define the complex transverse profile of a Gaussian laser.
+
+        **In the focal plane** (:math:`z=z_f`), the profile translates to a
+        laser with a transverse electric field:
+
+        .. math::
+
+            E(x,y,z=z_f) \propto \exp\left( -\\frac{r^2}{w_0^2} \\right)
+
+        where :math:`w_0` is the laser waist and :math:`r = \sqrt{x^2 + y^2}`.
+
+        Note that the peak amplitude of the profile is unity at the focus.
+
+        Parameters
+        ----------
+
+        waist: float (in meter)
+            Laser waist at the focal plane, defined as :math:`w_0` in the
+            above formula.
+
+        zf: float (in meter), optional
+            The position of the focal plane (in the lab frame).
+            If ``zf`` is not provided, the code assumes that ``zf=0.``.
+
+        lambda0: float (in meter), optional
+            The wavelength of the laser (in the lab frame), defined as
+            :math:`\\lambda_0` in the above formula.
+            Default: 0.8 microns (Ti:Sapph laser).
+
+        propagation_direction: int, optional
+            Indicates in which direction the laser propagates.
+            This should be either 1 (laser propagates towards positive z)
+            or -1 (laser propagates towards negative z).
+        """
+        # Initialize propagation direction and mark the profile as GPU capable
+        LaserTransverseProfile.__init__(self, propagation_direction,
+                                        gpu_capable=True)
+
+        # Wavevector and Rayleigh length
+        k0 = 2 * np.pi / lambda0
+        zr = 0.5 * k0 * waist ** 2
+        # Store the parameters
+        self.k0 = k0
```

Comment:
It seems that we do not need to store `k0` here.
```suggestion
        self.k0 = k0
```